packages/outline-view/src/browser/outline-view-widget.ts

```diff
@@ -0,0 +1,102 @@
+/*
+* Copyright (C) 2017 TypeFox and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+import { injectable, inject } from 'inversify';
+import { OutlineSymbolInformationNode, OutlineViewManager } from './outline-view-manager';
+import {
+    TreeWidget,
+    ITreeNode,
+    NodeProps,
+    ITreeModel,
+    ISelectableTreeNode,
+    TreeProps,
+    ContextMenuRenderer,
+    TreeModel,
+    IExpandableTreeNode
+} from "@theia/core/lib/browser";
+import { h } from "@phosphor/virtualdom/lib";
+import { Message } from '@phosphor/messaging';
+import { CommandRegistry } from '@theia/core';
+
+@injectable()
+export class OutlineViewWidget extends TreeWidget {
+
+    constructor(
+        @inject(TreeProps) protected readonly treeProps: TreeProps,
+        @inject(TreeModel) model: TreeModel,
+        @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer,
+        @inject(OutlineViewManager) protected readonly outlineViewManager: OutlineViewManager,
+        @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry
+    ) {
+        super(treeProps, model, contextMenuRenderer);
+
+        this.id = 'outline-view';
+        this.title.label = 'Outline';
+        this.addClass('theia-outline-view');
+    }
+
+    protected onAfterHide() {
+        this.outlineViewManager.isOpen = false;
+    }
+
+    protected onAfterShow() {
+        this.outlineViewManager.isOpen = true;
+    }
+
+    protected onUpdateRequest(msg: Message): void {
+        if (!this.model.selectedNode && ISelectableTreeNode.is(this.model.root)) {
+            this.model.selectNode(this.model.root);
+        }
+        super.onUpdateRequest(msg);
+    }
+
+    protected renderTree(model: ITreeModel): h.Child {
+        return super.renderTree(model);
+    }
+
+    protected decorateCaption(node: ITreeNode, caption: h.Child, props: NodeProps): h.Child {
+        if (OutlineSymbolInformationNode.is(node)) {
+            const icon = h.span({ className: "symbol-icon " + node.iconClass });
+            const theTree = h.div({
+                onclick: () => {
+                    this.commandRegistry.executeCommand(node.onClick, node);
```

Comment:
Just asking, no need to change. Why should it be implemented as a command?  Why not `this.manager.fireOpen` or `this.manager.fireSelect` and use events to notify monaco impl? Then you don't need `onOpen` and `onSelect` in nodes. I thought commands should be executable by users, not to be used for internal event emitting.

Reply:
:) good idea